src/lavinmq/vhost.cr

```diff
@@ -329,11 +331,13 @@ module LavinMQ
         when AMQP::Frame::Queue::Bind
           x = @exchanges[f.exchange_name]? || return false
           q = @queues[f.queue_name]? || return false
+          return false if x.has_binding?(q, f.routing_key, f.arguments.to_h)
```

Comment:
Cant Exchange#bind return false if the binding already existed? Now you do f.arguments.to_h twice and introduce a new method that might not be needed

Reply:
Yes, that was my initial thought. Dunno why I didn't go with that.